### PR TITLE
test: add backend e2e test for service

### PR DESCRIPTION
**Pull request by idbeta**
Please answer these questions before submitting a pull request

- Why submit this pull request?
- [x] E2E test
- [ ] New feature provided
- [ ] Improve performance

- Related issues

### CODE REVIEW

**Comment by idbeta** [comment]
ping @nic-chen @membphis

**Comment by idbeta** [comment]
ping @juzhiyuan Please remove the `Draft` tag of this PR.

File: api/test/e2e/service_test.go [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+			Headers:      map[string]string{"Authorization": token},
+			ExpectStatus: http.StatusOK,
+			Sleep:        sleepTime,
+		},
+	}
+	for _, tc := range tests {
+		testCaseCheck(tc, t)
+	}
+
+	// hit routes
```
Comment (membphis) [inline api/test/e2e/service_test.go:86]: bad comment

File: api/test/e2e/service_test.go [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+	}
+
+	// hit routes and check the response header
+	time.Sleep(sleepTime)
+	basepath := "http://127.0.0.1:9080"
+	request, _ := http.NewRequest("GET", basepath+"/server_port", nil)
+	request.Header.Add("Authorization", token)
+	resp, err := http.DefaultClient.Do(request)
+	if err != nil {
+		fmt.Printf("server not responding %s", err.Error())
```
Comment (membphis) [inline api/test/e2e/service_test.go:156]: I think we should throw an error here

**Review by membphis — CHANGES_REQUESTED** [review]

File: api/test/e2e/service_test.go (hunk 2 above)
Comment (idbeta) [reply api/test/e2e/service_test.go]: fixed.

File: api/test/e2e/service_test.go (hunk 1 above)
Comment (idbeta) [reply api/test/e2e/service_test.go]: fixed.

**Review by membphis — APPROVED** [review]

**Review by nic-chen — APPROVED** [review]

**Pull request merged**